azurerm/resource_arm_key_vault_access_policy_test.go

```diff
@@ -0,0 +1,397 @@
+					resource.TestCheckResourceAttr(resourceName2, "certificate_permissions.1", "delete"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMKeyVaultAccessPolicy_update(t *testing.T) {
+	rs := acctest.RandString(6)
+	resourceName := "azurerm_key_vault_access_policy.test"
+	preConfig := testAccAzureRMKeyVaultAccessPolicy_basic(rs, testLocation())
+	postConfig := testAccAzureRMKeyVaultAccessPolicy_update(rs, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		// Using the KeyVaultDestroy checker since that ensures the policy is gone.
+		CheckDestroy: testCheckAzureRMKeyVaultDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: preConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMKeyVaultAccessPolicyExists(resourceName, "object_id", "application_id"),
+					resource.TestCheckResourceAttr(resourceName, "key_permissions.0", "get"),
+					resource.TestCheckResourceAttr(resourceName, "secret_permissions.0", "get"),
+					resource.TestCheckResourceAttr(resourceName, "secret_permissions.1", "set"),
+				),
+			},
+			{
+				Config: postConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMKeyVaultAccessPolicyExists(resourceName, "object_id", "application_id"),
+					resource.TestCheckResourceAttr(resourceName, "key_permissions.0", "list"),
+					resource.TestCheckResourceAttr(resourceName, "key_permissions.1", "encrypt"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMKeyVaultAccessPolicy_policyRemoved(t *testing.T) {
+	rs := acctest.RandString(6)
+	policyResourceName := "azurerm_key_vault_access_policy.test"
+	vaultResourceName := "azurerm_key_vault.test"
+	preConfig := testAccAzureRMKeyVaultAccessPolicy_basic(rs, testLocation())
+	postConfig := testAccAzureRMKeyVaultAccessPolicy_policyRemoved(rs, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		// Using the KeyVaultDestroy checker since that ensures the policy is gone.
+		CheckDestroy: testCheckAzureRMKeyVaultDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: preConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMKeyVaultAccessPolicyExists(policyResourceName, "object_id", "application_id"),
+					resource.TestCheckResourceAttr(policyResourceName, "key_permissions.0", "get"),
+					resource.TestCheckResourceAttr(policyResourceName, "secret_permissions.0", "get"),
+					resource.TestCheckResourceAttr(policyResourceName, "secret_permissions.1", "set"),
+				),
+			},
+			{
+				Config: postConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMKeyVaultAccessPolicyMissing(vaultResourceName, "tags.policy_object_id"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMKeyVaultAccessPolicyExists(name string, policyObjectTag string, policyApplicationTag string) resource.TestCheckFunc {
```

Comment:
Should `policyApplicationTag` be `policyApplicationId`

Reply:
It isn't the "Id" though the the index of the id in the resource object.  This was a suggestion from @tombuildsstuff to pass this through here.  I will rename it if requested.